### ISSUE TITLE
Code completion bug: Function argument suggestion for R functions does not work anymore in v1.3

### ISSUE THREAD

**Issue by tungmilan** [rstudio/rstudio#7060]
Hello,

I noticed that in v1.3, function argument suggestion no longer works after the first argument of a function. Could you check if it's the case for you guys too? Thanks!

### System details

    RStudio Edition : Desktop
    RStudio Version : Version 1.3.1008
    OS Version      : Windows 10 x64 (build 16299)
    R Version       : R version 4.0.1 (2020-06-06)

### Steps to reproduce the problem

```r
library(tidyr)

iris_long <- iris %>% 
  pivot_longer(names_to = 'variable',)
```

### Describe the problem in detail

After the first comma `,`, function argument suggestion stopped working 

![](https://i.imgur.com/u9X3iV7.png)

### Describe the behavior you expected

argument suggestion should work after each comma `,`  (like in v1.2.5042)

![](https://i.imgur.com/2LCkpZN.png)

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
I see the same behavior with RStudio 1.2.5042, so this doesn't appear to be a regression.

**Comment by tungmilan** [comment]
Thanks for checking @kevinushey ! So it was because of some packages got updated? Here is my session info that still has argument suggestion working

````
- Packages -------------------------------------------------------------------------------------------------------------
 package     * version     date       lib source                          
 assertthat    0.2.1       2019-03-21 [1] CRAN (R 3.6.1)                  
 backports     1.1.6       2020-04-05 [1] CRAN (R 3.6.2)                  
 callr         3.4.3       2020-03-28 [1] CRAN (R 3.6.3)                  
 cli           2.0.2       2020-02-28 [1] CRAN (R 3.6.3)                  
 crayon        1.3.4       2017-09-16 [1] CRAN (R 3.6.1)                  
 desc          1.2.0       2018-05-01 [1] CRAN (R 3.6.1)                  
 devtools      2.3.0       2020-04-10 [1] CRAN (R 3.6.2)                  
 digest        0.6.25      2020-02-23 [1] CRAN (R 3.6.2)                  
 dplyr         0.8.99.9002 2020-04-23 [1] Github (tidyverse/dplyr@b6e5c5c)
 ellipsis      0.3.0       2019-09-20 [1] CRAN (R 3.6.1)                  
 evaluate      0.14        2019-05-28 [1] CRAN (R 3.6.1)                  
 fansi         0.4.1       2020-01-08 [1] CRAN (R 3.6.2)                  
 fortunes      1.5-4       2016-12-29 [1] CRAN (R 3.6.0)                  
 fs            1.4.1       2020-04-04 [1] CRAN (R 3.6.2)                  
 generics      0.0.2       2018-11-29 [1] CRAN (R 3.6.1)                  
 glue          1.4.0.9000  2020-04-23 [1] Github (tidyverse/glue@a3375fe) 
 htmltools     0.4.0       2019-10-04 [1] CRAN (R 3.6.1)                  
 knitr         1.28        2020-02-06 [1] CRAN (R 3.6.2)                  
 lifecycle     0.2.0       2020-03-06 [1] CRAN (R 3.6.3)                  
 magrittr      1.5         2014-11-22 [1] CRAN (R 3.6.1)                  
 memoise       1.1.0       2017-04-21 [1] CRAN (R 3.6.1)                  
 pillar        1.4.3.9001  2020-04-04 [1] Github (r-lib/pillar@52b4503)   
 pkgbuild      1.0.6       2019-10-09 [1] CRAN (R 3.6.1)                  
 pkgconfig     2.0.3       2019-09-22 [1] CRAN (R 3.6.1)                  
 pkgload       1.0.2       2018-10-29 [1] CRAN (R 3.6.1)                  
 prettyunits   1.1.1       2020-01-24 [1] CRAN (R 3.6.2)                  
 processx      3.4.2       2020-02-09 [1] CRAN (R 3.6.2)                  
 ps            1.3.2       2020-02-13 [1] CRAN (R 3.6.2)                  
 purrr         0.3.4       2020-04-17 [1] CRAN (R 3.6.3)                  
 R6            2.4.1       2019-11-12 [1] CRAN (R 3.6.1)                  
 Rcpp          1.0.4.6     2020-04-09 [1] CRAN (R 3.6.3)                  
 remotes       2.1.1       2020-02-15 [1] CRAN (R 3.6.2)                  
 rlang         0.4.5.9000  2020-04-14 [1] Github (r-lib/rlang@a90b04b)    
 rmarkdown     2.1         2020-01-20 [1] CRAN (R 3.6.2)                  
 rprojroot     1.3-2       2018-01-03 [1] CRAN (R 3.6.1)                  
 rstudioapi    0.11        2020-02-07 [1] CRAN (R 3.6.2)                  
 sessioninfo   1.1.1       2018-11-05 [1] CRAN (R 3.6.1)                  
 testthat      2.3.2       2020-03-02 [1] CRAN (R 3.6.3)                  
 tibble        3.0.1       2020-04-20 [1] CRAN (R 3.6.3)                  
 tidyr       * 1.0.2.9000  2020-04-24 [1] Github (tidyverse/tidyr@6d889d0)
 tidyselect    1.0.0       2020-01-27 [1] CRAN (R 3.6.2)                  
 usethis       1.6.0       2020-04-09 [1] CRAN (R 3.6.3)                  
 vctrs         0.2.99.9011 2020-04-23 [1] Github (r-lib/vctrs@bf44281)    
 withr         2.2.0       2020-04-20 [1] CRAN (R 3.6.3)                  
 xfun          0.13        2020-04-13 [1] CRAN (R 3.6.2)                  
 yaml          2.2.1       2020-02-01 [1] CRAN (R 3.6.2)
````

**Comment by bdhhall** [comment]
I experience the same issue (just found this before I opened my own issue), and can add some additional checking I did in hopes of helping to narrow down the source of the problem.

1. Testing this in RStudio Cloud, using R version 4.0.0 and 3.6.3, the problem is actually worse on Cloud, but identical regardless of R version. Example code: `tibble(a = 1:5, b = a * 2) %>% pivot_longer(data = ., data)`

If you erase "data" as the second argument of pivot-longer above, then press Tab, it'll just insert "data", no autocomplete options at all. What it inserts seems to just be the the first param name you give to the function, regardless of what it is, always.

On RStudio Desktop (Windows 10) Version 1.3.959 (R version 3.6.1), pressing Tab will instead give you the parameter-less autocomplete options mentioned by OP.

I tested this after updating from RStudio Desktop 1.1.x (don't recall the exact minor version number), and got the same behavior. I updated tidyverse/tidyr and RStudio to see if that might change anything, but no dice.

2. This bug seems oddly specific to certain functions. For example, in tidyr this exact code completion bug occurs with pivot_longer(), pivot_wider(), and gather_(), but not with pivot_longer_spec(), pivot_wider_spec(), spread(), gather(), etc.

3. There is a second part of this bug, possibly related, involving the additional help information provided next to the autocomplete, which occurs even for functions where the argument suggestion works. Examples below. I'm not sure if this should be labeled its own bug, but it seems so related I'll leave it up to someone else to decide if a separate issue should be posted.

Working, expected behavior for first param of gather(): 
![gather_working_first_param](https://user-images.githubusercontent.com/25888168/85628195-b5f1bb80-b63d-11ea-9af0-6dedadf0e69e.png)

Broken, unexpected behavior for second param of gather():
![gather_broken_second_param](https://user-images.githubusercontent.com/25888168/85628228-c73ac800-b63d-11ea-85a9-ee8d1e63c3a7.png)

Note the yellow documentation box appears just fine for the first parameter, but never appears for any additional parameter. But this is not true of all functions. For example, it works just fine for pivot_longer_spec(), shown below.

Working, expected behavior for first param of pivot_longer_spec():
![pivot-long-spec-working_first_param](https://user-images.githubusercontent.com/25888168/85628767-ad4db500-b63e-11ea-9a31-9316093e711e.png)

Working, expected behavior for second param of pivot_longer_spec():
![pivot-long-spec-working_second_param](https://user-images.githubusercontent.com/25888168/85628775-b179d280-b63e-11ea-8411-90b3db3656d6.png)

Testing a bit further, I realize many functions seem to exhibit this secondary autocomplete issue. For example, I found it broken with functions: ggplot, ggplot_add, filter, mutate, group_by, and gather. Yet this feature works fine for pivot_wider_spec, pivot_longer_spec, seq.int, modelr::add_predictions, and tidyquant::COUNT_DAYS - just to randomly try a few different functions. 

So these bugs don't seem strictly related to any single package, as far as I can figure. 

4. Finally, I've tried new projects, a new project on RStudio Cloud, deleting history/cache, updating packages, etc. - no change in behavior.

TLDR; there are 1-2 bugs in autocomplete functionality that are specific to 2nd-and-beyond parameters of functions, but this is not universal to all functions, nor to all functions within any given package (some functions in tidyr work fine, while others don't - see pivot_longer vs pivot_longer_spec). Testing with RStudio Desktop 1.1.x and 1.3.x shows same or similar odd behavior, while testing with RStudio Cloud shows worse behavior (no autocomplete options, inserts first param name instead). R version does not appear to matter, either.

**Comment by mdancho84** [comment]
I'm experiencing this behavior as well. It seems like autocomplete isn't working as anticipated. After the 2nd argument, tab completion goes away. 

![image](https://user-images.githubusercontent.com/13734662/86143933-0a27f000-bac3-11ea-9a45-561c5e1cee75.png)

**Comment by kevinushey** [comment]
Should be fixed via https://github.com/rstudio/rstudio/pull/7906. Thanks for reporting!

**Comment by tungmilan** [comment]
Thanks @kevinushey ! I can confirm that `v1.4.890` works now

**Comment by rich-rstudio** [comment]
verified on Version 1.4.938

**Comment by calpan** [comment]
On v1.4.904, the following error appears sometimes upon hitting Tab to trigger the autocompletion:

Error in if (length(new$fguess) && new$fguess != "") old$fguess <- new$fguess : 
  missing value where TRUE/FALSE needed

![2020-10-16_22-48-16](https://user-images.githubusercontent.com/806424/96329471-07b1bd80-1002-11eb-837b-ef0bea30c95e.gif)

I'm not sure if this behavior is new, but I hadn't seen it before.

**Comment by tungmilan** [comment]
Confirmed the error message that @calpan encountered on `1.4.949`, Windows 10 OS.
After typing `by` inside `left_join`, pressing Tab will generate the message

```
Error in if (length(new$fguess) && new$fguess != "") old$fguess <- new$fguess : 
  missing value where TRUE/FALSE needed
```

```
library(dplyr)

mtcars %>% 
  left_join(mtcars,
            by)
```

**Comment by kevinushey** [comment]
Thanks @tungmilan! I've reproduced as well with your help and am taking a look.

**Comment by tungmilan** [comment]
Thank you @kevinushey ! Just tested and we are good now

**Comment by kevinushey** [comment]
Awesome, thank you again for testing and reporting back!

**Comment by iamnicogomez** [comment]
Is this fix for exclusively the Windows 10 OS RStudio release? I can't tell if I'm the only one reporting this, but I have a similar/identical problem on Mac OS 10.15.6, RStudio Version 1.3.1093. I don't always use argument completion, but with tidyr::pivot_x() it is very useful.

Similarly, an empty function does in fact show named arguments:
![image](https://user-images.githubusercontent.com/39886182/97114071-b91cb700-16c4-11eb-96d6-3346c45d4a41.png)


But if you type anything at all, it displays environmental variables:
(passing '.' to 'data')
![image](https://user-images.githubusercontent.com/39886182/97114097-d5b8ef00-16c4-11eb-8e23-bbcc642b7257.png)
(passing a column to 'col')
![image](https://user-images.githubusercontent.com/39886182/97114133-07ca5100-16c5-11eb-8e35-59ee8be1c3e2.png)


I'm not sure how to provide a reproducible version of this bug here, but here's my session info:

```
sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin19.5.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /usr/local/Cellar/openblas/0.3.10_1/lib/libopenblasp-r0.3.10.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2     purrr_0.3.4     readr_1.4.0     tidyr_1.1.2     tibble_3.0.3    ggplot2_3.3.2   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] bitops_1.0-6                matrixStats_0.57.0          fs_1.5.0                    lubridate_1.7.9             bit64_4.0.5                 progress_1.2.2             
 [7] httr_1.4.2                  GenomeInfoDb_1.24.2         tools_4.0.2                 backports_1.1.10            R6_2.4.1                    DBI_1.1.0                  
[13] BiocGenerics_0.34.0         colorspace_1.4-1            withr_2.3.0                 tidyselect_1.1.0            prettyunits_1.1.1           bit_4.0.4                  
[19] curl_4.3                    compiler_4.0.2              cli_2.0.2                   rvest_0.3.6                 Biobase_2.48.0              xml2_1.3.2                 
[25] DelayedArray_0.14.1         rtracklayer_1.48.0          scales_1.1.1                askpass_1.1                 rappdirs_0.3.1              digest_0.6.25              
[31] Rsamtools_2.4.0             XVector_0.28.0              pkgconfig_2.0.3             dbplyr_1.4.4                BSgenome_1.56.0             rlang_0.4.8                
[37] readxl_1.3.1                rstudioapi_0.11             RSQLite_2.2.1               generics_0.0.2              jsonlite_1.7.1              BiocParallel_1.22.0        
[43] RCurl_1.98-1.2              magrittr_1.5                GenomeInfoDbData_1.2.3      Matrix_1.2-18               Rcpp_1.0.5                  munsell_0.5.0              
[49] S4Vectors_0.26.1            fansi_0.4.1                 clipr_0.7.1                 lifecycle_0.2.0             stringi_1.5.3               SummarizedExperiment_1.18.2
[55] zlibbioc_1.34.0             BiocFileCache_1.12.1        grid_4.0.2                  blob_1.2.1                  parallel_4.0.2              crayon_1.3.4               
[61] lattice_0.20-41             Biostrings_2.56.0           haven_2.3.1                 hms_0.5.3                   pillar_1.4.6                GenomicRanges_1.40.0       
[67] biomaRt_2.45.6              stats4_4.0.2                reprex_0.3.0                XML_3.99-0.5                glue_1.4.2                  modelr_0.1.8               
[73] vctrs_0.3.4                 cellranger_1.1.0            gtable_0.3.0                openssl_1.4.3               assertthat_0.2.1            xfun_0.18                  
[79] broom_0.7.1                 GenomicAlignments_1.24.0    tinytex_0.26                AnnotationDbi_1.50.3        memoise_1.1.0               IRanges_2.22.2             
[85] ellipsis_0.3.1
```

**Comment by tungmilan** [comment]
@iamnicogomez: you need to update your RStudio version to the latest preview or daily version

**Comment by chrisvwn** [comment]
I am editing a package and when I try the tab expansion within the code editor (on console this is expected behavior) on an unexported function I do not get a list of parameters. I am wondering if this has changed? I seem to remember getting the parameter lists as this makes calling functions within the package hence coding the package so much easier. Tab expansion with exported functions seems to work fine.

Rstudio version: 1.3.1093

My sessionInfo:
```
R version 4.0.3 (2020-10-10)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 33 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Rnightlights_0.2.5.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        compiler_4.0.3    pillar_1.4.6      plyr_1.8.6        bitops_1.0-6     
 [6] iterators_1.0.12  tools_4.0.3       digest_0.6.27     lattice_0.20-41   viridisLite_0.3.0
[11] jsonlite_1.7.1    lifecycle_0.2.0   tibble_3.0.4      gtable_0.3.0      pkgconfig_2.0.3  
[16] rlang_0.4.8       foreach_1.5.0     rstudioapi_0.13   rgdal_1.5-18      yaml_2.2.1       
[21] httr_1.4.2        stringr_1.4.0     dplyr_1.0.2       htmlwidgets_1.5.1 generics_0.1.0   
[26] vctrs_0.3.5       grid_4.0.3        tidyselect_1.1.0  glue_1.4.2        data.table_1.13.2
[31] R6_2.5.0          plotly_4.9.2.1    sp_1.4-4          tidyr_1.1.2       ggplot2_3.3.2    
[36] reshape2_1.4.4    purrr_0.3.4       magrittr_2.0.1    scales_1.1.1      codetools_0.2-16 
[41] settings_0.2.4    ellipsis_0.3.1    htmltools_0.5.0   colorspace_2.0-0  stringi_1.5.3    
[46] lazyeval_0.2.2    RCurl_1.98-1.2    munsell_0.5.0     crayon_1.3.4  
```
